### PR TITLE
Fix a number of build warnings

### PR DESCRIPTION
**Pull request by maliroteh**
@kchitalia @smcnulty-sfdc @wpeter-sfdc @jbernoudy @jhorstsf

### CODE REVIEW

**Comment by kchitalia** [comment]
:+1: Looks good to me.

File: SalesforceSDK/Salesforce.SDK.App/Hybrid/HybridMainPage.cs [hunk 1]
```diff
@@ -183,11 +183,12 @@ protected void OnResumeLoggedInNotLoaded()
                     // Offline
                 else
                 {
+                    // TODO: FIXME
                     // Has cached version
-                    if (false /* FIXME */)
-                    {
-                        Log("Success:Loading cached version of remote application because offline");
-                    }
+                    //if (false)
```
Comment (bhariharan) [inline SalesforceSDK/Salesforce.SDK.App/Hybrid/HybridMainPage.cs]: Why's this commented out? Can we remove this?

File: SalesforceSDK/Salesforce.SDK.SmartStore/Source/Store/SmartStore.cs [hunk 2]
```diff
@@ -695,23 +695,20 @@ private object GetObject(SQLiteStatement statement, int position)
             {
                 return statement.GetText(position);
             }
-            catch (Exception e)
-            {
-            }
+            catch (Exception) { }
```
Comment (bhariharan) [inline SalesforceSDK/Salesforce.SDK.SmartStore/Source/Store/SmartStore.cs]: Should we add log statements when an exception is caught here?

File: SalesforceSDK/Salesforce.SDK.App/Hybrid/HybridMainPage.cs (hunk 1 above)
Comment (maliroteh) [inline SalesforceSDK/Salesforce.SDK.App/Hybrid/HybridMainPage.cs]: This is for caching and we don't have that implemented yet. @kchitalia will be working on fixing up HybridMainPage and cleaning up as needed. I just left this TODO here so we don't forget about this.

File: SalesforceSDK/Salesforce.SDK.SmartStore/Source/Store/SmartStore.cs (hunk 2 above)
Comment (maliroteh) [inline SalesforceSDK/Salesforce.SDK.SmartStore/Source/Store/SmartStore.cs]: We're using the Try-Catch approach here to determine if the object is Text/Int/Float. I don't think there is a need to log anything here every time the method is called. The log statement will not provide any real added value. But there may be a better approach than Try-Catch. @kchitalia will be looking at this file to clean up. I think we can get the data type using statement.DataType and using a switch statement we can determine which of GetText/GetInteger/GetFloat to call without needing any Try-Catch

Comment (bhariharan) [inline SalesforceSDK/Salesforce.SDK.SmartStore/Source/Store/SmartStore.cs]: Makes sense. Thanks @maliroteh.

**Pull request merged**